### ISSUE TITLE
An assembly specified in the application dependencies manifest was not found

### ISSUE THREAD

**Issue by mjnorman** [cloudfoundry/dotnet-core-buildpack#192]
What version of Cloud Foundry and CF CLI are you using? (i.e. What is the output of running `cf curl /v2/info && cf version`?
cf version 6.35.0+5c7de3e32.2018-03-08

What version of the buildpack you are using?
Dotnet-Core Buildpack version 2.1.4

If you were attempting to accomplish a task, what was it you were attempting to do?
publish app to CF

What did you expect to happen?
no errors

What was the actual behavior?
Received Error: 
```
2018-08-08T08:51:27.703-05:00 [APP/PROC/WEB/0] [ERR] An assembly specified in the application dependencies manifest (CallbacksCore.Web.deps.json) was not found:
2018-08-08T08:51:27.703-05:00 [APP/PROC/WEB/0] [ERR] package: 'Microsoft.ApplicationInsights.AspNetCore', version: '2.1.1'
2018-08-08T08:51:27.703-05:00 [APP/PROC/WEB/0] [ERR] path: 'lib/netstandard1.6/Microsoft.ApplicationInsights.AspNetCore.dll'
2018-08-08T08:51:27.703-05:00 [APP/PROC/WEB/0] [ERR] This assembly was expected to be in the local runtime store as the application was published using the following target manifest files:
2018-08-08T08:51:27.703-05:00 [APP/PROC/WEB/0] [ERR] aspnetcore-store-2.0.0-linux-x64.xml;aspnetcore-store-2.0.0-osx-x64.xml;aspnetcore-store-2.0.0-win7-x64.xml;aspnetcore-store-2.0.0-win7-x86.xml
```

Please confirm where necessary:
* [x] I have included a log output
* [x] My log includes an error message
* [x] I have included steps for reproduction

I know there is a workaround to put 
```
  <PropertyGroup>
    <PublishWithAspNetCoreTargetManifest>false</PublishWithAspNetCoreTargetManifest>
  </PropertyGroup>
```
in your csproj, but I don't understand why this works.  Supposedly, the issue is caused by the fact that the application assumes that the files needed will already be installed via SDK, but for some reason they are missing.  However I can see from log output that the SDK specified in global.json is installed by the buildpack.  Is this particular file just missing?

```
   -----> Installing dotnet 2.1.302
          Download [https://buildpacks.cloudfoundry.org/dependencies/dotnet/dotnet.2.1.302.linux-amd64-cflinuxfs2-17f2d281.tar.xz]
```

**Comment by astrieanna** [comment]
Hi @mjnorman,

Are you pushing a framework-dependent or an unpublished dotnet app?

Where is this manifest file coming from? Can you show us the `aspnetcore-store-2.0.0-linux-x64.xml` file?

**Comment by astrieanna** [comment]
Could we see the app? It might be easier for us to reproduce this behavior to debug it.

**Comment by mjnorman** [comment]
We host on a private corporate github, but I can clone it here and remove the connection strings (sensitive data), would that be sufficient?

**Comment by mjnorman** [comment]
I was wrong we do perform a dotnet publish during the pipeline, just a standard "dotnet publish"

**Comment by astrieanna** [comment]
Is your pipeline possibly running another tool that generates the manifest file? What arguments is `dotnet publish` run with?

**Comment by mjnorman** [comment]
No arguments.  It is just a straight dotnet publish.

**Comment by astrieanna** [comment]
It sounds like, when your pipeline runs `dotnet publish`, it is publishing against a custom store manifest (`aspnetcore-store-2.0.0-linux-x64.xml`), which is not a manifest that we have published.

A custom store manifest tells dotnet what assemblies will be available in a particular target environment. What the the dotnet buildpack makes available doesn't match the list in this manifest file.

The main question here is where this manifest file that is being used by your pipeline is coming from? And how you can make your pipeline not use that file (which is what your workaround is doing).

**Comment by sclevine** [comment]
I believe `aspnetcore-store-2.0.0-linux-x64.xml` is the default manifest that is shipped with the 2.0.0 (or maybe 2.0.x?) version of .NET Core SDK. If you don't want to use the `<PublishWithAspNetCoreTargetManifest>false</PublishWithAspNetCoreTargetManifest>` workaround, you can control the SDK version that the buildpack pulls assemblies from via the `sdk` setting for `dotnet-core` in buildpack.yml. Try:
```
dotnet-core:
  sdk: 2.0.x
```
Alternatively, update your pipeline to use .NET Core 2.1 to match the version provided by the buildpack.

We plan to streamline this workflow in version 2.2.0 of the buildpack by publishing a custom aspnetcore-store for each supported line of .NET Core.

**Comment by mjnorman** [comment]
I’m probably confused, I don’t know anyone who isn’t by dotnet core versioning these days. 

But if the buildpack release notes says that 2.0.0 is supported, unless I’m misunderstanding what that means, why wouldn’t that manifest be included?

**Comment by sclevine** [comment]
The list of supported dependencies in the latest version of the buildpack is here:
http://buildpacks.cloudfoundry.org/#/buildpacks/dotnet/v2.1.4

The buildpack includes the 2.0.3 SDK, but if you push a pre-published app, it will provide the ASP.NET Core assemblies that were released with the latest SDK version, which is 2.1.302. This is because the buildpack bundles the ASP.NET Core assemblies with the SDK tools. (We plan to change this is the future.)

So to make this work for now, you can use buildpack.yml to tell the buildpack which SDK version to pull the assemblies from.

**Comment by jfeeny** [comment]
@mjnorman We'll go ahead and close this issue now as you have a workaround. Also, this behavior will likely be fixed without a workaround upon completion of https://www.pivotaltracker.com/story/show/159044086.
